File: solutions/kotlin/sum-of-multiples/3/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {

    fun sum(factors: Set<Int>, limit: Int): Int {
        var ans = 0
        for (i in 1 until limit) {
            for (f in factors) {
                if (f != 0 && i % f == 0) {
                    ans += i
                    break
                }
            }
        }
        return ans
    }
}

File: solutions/csharp/sum-of-multiples/3/SumOfMultiples.cs
public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        bool[] a = Enumerable.Repeat(false, max).ToArray();
        foreach (var x in multiples) {
            var y = x;
            while (y != 0 && y < max)
            {
                a[y] = true;
                y += x;
            }
        }
        var ans = 0;
        for (var i = 0; i < max; i++)
        {
            if (a[i])
            {
                ans += i;
            }
        }
        return ans;
    }
}